pkg/controllers/provisioning/provisioner.go

```diff
@@ -345,6 +345,8 @@ func (p *Provisioner) Schedule(ctx context.Context) (scheduler.Results, error) {
 		}
 		return scheduler.Results{}, fmt.Errorf("creating scheduler, %w", err)
 	}
+	// Emit a metric for when we first acknowledged the pod for scheduling
+	p.cluster.SetPodAckTime(pendingPods...)
```

Comment:
This emits when we first see the pod but doesn't signal whether the pod can actually schedule -- aren't the metrics going to be affected by whether we _can_ actually schedule the pod? Seems like we need a metric that marks down how long it took for us to go from pod seen to considered by the scheduler (either able to be scheduled or not) and another metric which tells us how long it took us to go from determined that we can schedule it to actually getting the pod bound.

Reply:
That's a good point. So what we really need is to track the state transitions between four points and emit metrics from `1 -> 2`, `2->3`, and `2-> 4`.
1. Pod created
2. When did we first see the pod in our scheduling loop
3. When did we first think that we could fit the pod on a node (theoretical or not)
4. When did the pod actually bind

Then we track metrics of 
i. Ack (emitted when ACK'ing the pods on this line in code)
  a. Per-pod ack Gauge (1 -> now)
  b. Histogram ack metric (1 -> 2)
ii. Schedulable (should be from when it was ACK'd, not created, emitted when we actually successfully node.add(pod) in the scheduling loop)
  a. Per-pod schedulable Gauge (2 -> now) 
  b. Histogram schedulable (2 -> 3) 
iii. Scheduled (same as bound, but since it was ACK'd, not created) (emitted when the pod actually binds)
  a. Per-pod scheduled Gauge (2 -> now)
  b. Histogram scheduled (2 -> 4)

Reply:
Yep, fully aligned with this 👍🏼